File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternative/ContractAlternative.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace ContractsAlternative
    {
        public class ContractAlternative
        {
            private DataTypeValidator _dataTypeValidator;

            public ContractAlternative()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _ContractID = 0;

            public int ContractID
            {
                get
                {
                    return _ContractID;
                }
            }

            public object SetContractID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractID", value);
                }
            }

            private int _CustomerID = 0;

            public int CustomerID
            {
                get
                {
                    return _CustomerID;
                }
            }

            public object SetCustomerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_CustomerID", value);
                }
            }

            private string _ContractReference = string.Empty;

            public string ContractReference
            {
                get
                {
                    return _ContractReference;
                }
            }

            public object SetContractReference
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractReference", value);
                }
            }

            private DateTime _ContractStartDate = default;

            public DateTime ContractStartDate
            {
                get
                {
                    return _ContractStartDate;
                }

                set
                {
                    _ContractStartDate = value;
                }
            }

            private DateTime _ContractEndDate = default;

            public DateTime ContractEndDate
            {
                get
                {
                    return _ContractEndDate;
                }

                set
                {
                    _ContractEndDate = value;
                }
            }

            private int _ContractStatusID = 0;

            public int ContractStatusID
            {
                get
                {
                    return _ContractStatusID;
                }
            }

            public object SetContractStatusID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractStatusID", value);
                }
            }

            private double _ContractPrice = 0;

            public double ContractPrice
            {
                get
                {
                    return _ContractPrice;
                }
            }

            public object SetContractPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractPrice", value);
                }
            }

            private int _QuoteContractID = 0;

            public int QuoteContractID
            {
                get
                {
                    return _QuoteContractID;
                }
            }

            public object SetQuoteContractID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteContractID", value);
                }
            }

            private int _InvoiceFrequencyID = 0;

            public int InvoiceFrequencyID
            {
                get
                {
                    return _InvoiceFrequencyID;
                }
            }

            public object SetInvoiceFrequencyID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceFrequencyID", value);
                }
            }

            private DateTime _FirstInvoiceDate;

            public DateTime FirstInvoiceDate
            {
                get
                {
                    return _FirstInvoiceDate;
                }

                set
                {
                    _FirstInvoiceDate = value;
                }
            }

            private int _InvoiceAddressTypeID;

            public int InvoiceAddressTypeID
            {
                get
                {
                    return _InvoiceAddressTypeID;
                }
            }

            public object SetInvoiceAddressTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceAddressTypeID", value);
                }
            }

            private int _InvoiceAddressID = 0;

            public int InvoiceAddressID
            {
                get
                {
                    return _InvoiceAddressID;
                }
            }

            public object SetInvoiceAddressID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceAddressID", value);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternative/ContractAlternative.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ContractsAlternative

        Public Class ContractAlternative

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Contract Properties"


            Private _ContractID As Integer = 0
            Public ReadOnly Property ContractID() As Integer
                Get
                    Return _ContractID
                End Get
            End Property
            Public WriteOnly Property SetContractID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractID", Value)
                End Set
            End Property


            Private _CustomerID As Integer = 0
            Public ReadOnly Property CustomerID() As Integer
                Get
                    Return _CustomerID
                End Get
            End Property
            Public WriteOnly Property SetCustomerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_CustomerID", Value)
                End Set
            End Property


            Private _ContractReference As String = String.Empty
            Public ReadOnly Property ContractReference() As String
                Get
                    Return _ContractReference
                End Get
            End Property
            Public WriteOnly Property SetContractReference() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractReference", Value)
                End Set
            End Property


            Private _ContractStartDate As DateTime = Nothing
            Public Property ContractStartDate() As DateTime
                Get
                    Return _ContractStartDate
                End Get
                Set(ByVal Value As DateTime)
                    _ContractStartDate = Value
                End Set
            End Property


            Private _ContractEndDate As DateTime = Nothing
            Public Property ContractEndDate() As DateTime
                Get
                    Return _ContractEndDate
                End Get
                Set(ByVal Value As DateTime)
                    _ContractEndDate = Value
                End Set
            End Property


            Private _ContractStatusID As Integer = 0
            Public ReadOnly Property ContractStatusID() As Integer
                Get
                    Return _ContractStatusID
                End Get
            End Property
            Public WriteOnly Property SetContractStatusID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractStatusID", Value)
                End Set
            End Property


            Private _ContractPrice As Double = 0
            Public ReadOnly Property ContractPrice() As Double
                Get
                    Return _ContractPrice
                End Get
            End Property
            Public WriteOnly Property SetContractPrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractPrice", Value)
                End Set
            End Property


            Private _QuoteContractID As Integer = 0
            Public ReadOnly Property QuoteContractID() As Integer
                Get
                    Return _QuoteContractID
                End Get
            End Property
            Public WriteOnly Property SetQuoteContractID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteContractID", Value)
                End Set
            End Property


            Private _InvoiceFrequencyID As Integer = 0
            Public ReadOnly Property InvoiceFrequencyID() As Integer
                Get
                    Return _InvoiceFrequencyID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceFrequencyID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceFrequencyID", Value)
                End Set
            End Property


            Private _FirstInvoiceDate As DateTime
            Public Property FirstInvoiceDate() As DateTime
                Get
                    Return _FirstInvoiceDate
                End Get
                Set(ByVal Value As DateTime)
                    _FirstInvoiceDate = Value
                End Set
            End Property


            Private _InvoiceAddressTypeID As Integer
            Public ReadOnly Property InvoiceAddressTypeID() As Integer
                Get
                    Return _InvoiceAddressTypeID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceAddressTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceAddressTypeID", Value)
                End Set
            End Property


            Private _InvoiceAddressID As Integer = 0
            Public ReadOnly Property InvoiceAddressID() As Integer
                Get
                    Return _InvoiceAddressID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceAddressID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceAddressID", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

